kong/conf_loader/init.lua

```diff
@@ -660,6 +664,9 @@ local CONF_INFERENCES = {
 
   lmdb_environment_path = { typ = "string" },
   lmdb_map_size = { typ = "string" },
+
+  proxy_restricted_hosts = { typ = "array" },
```

Comment:
Do you think we need to at some point provide both allow and deny, if so perhaps proxy_allowed_hosts or proxy_hosts_allow and proxy_denied_hosts or proxy_hosts_deny?

Reply:
I like this idea. We could make them mutually-exclusive, and the presence of an "allowed" list could invert the logic to deny all except for those explicitly allowed.

I also think the naming of `proxy_denied_hosts`/`proxy_hosts_deny` is slightly less ambiguous when compared to `proxy_restricted_hosts`.

Reply:
It's a product decision to not add `proxy_allowed_hosts` now. 

I agree that `proxy_denied_hosts` is clearer, changed.